lib/system_win32.c

```diff
@@ -89,6 +95,17 @@ CURLcode Curl_win32_init(long flags)
   }
 #endif
 
+  s_hIpHlpApiDll = Curl_load_library(TEXT("iphlpapi.dll"));
+  if(s_hIpHlpApiDll) {
+    /* Get the address of the if_nametoindex function */
+    IF_NAMETOINDEX_FN pIfNameToIndex =
```

Comment:
I suggest assign directly to Curl_if_nametoindex. Other than that this looks fine to me. But don't ask me what's up with the segfaults in the --enable-ares build. I can't fathom how they could possibly be related. Please make sure to add `Fixes`/`Closes`/`Bug:`/`Ref:`/etc lines to the commit message before sending it upstream.